Bind to all interfaces

JesHansen opened the pull request:
When running under WSL2, binding only to localhost via the 127.0.0.1 address
only lets you access the server from within WSL2 itself. It does not allow you
to access the server from Windows.

With this change, you can access the web server from the WSL ip address. You can
find your WSL ip address with

```
$ ip a | grep eth0
4: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    inet 192.168.161.12/20 brd 192.168.175.255 scope global eth0
```

LukeMathWalker commented:
Conflicting feelings here: I intuitively feel that bindings to all network interfaces with 0.0.0.0 is a _bad default_.
In some environments is needed (e.g. WSL2 or Docker) but I would like to avoid opening the web server to traffic coming (potentially) from the internet as a default setting.
Maybe as a comment?

From chapter03-1 it becomes configurable, so it should be easier for people to customise it.

JesHansen commented:
> Conflicting feelings here: I intuitively feel that bindings to all network interfaces with 0.0.0.0 is a _bad default_.

I agree.

> In some environments is needed (e.g. WSL2 or Docker) but I would like to avoid opening the web server to traffic coming (potentially) from the internet as a default setting.
> Maybe as a comment?

This would be a great solution. Keep the defaults secure, but let people know that it could cause issues. That way people can be spared the debugging session

> From chapter03-1 it becomes configurable, so it should be easier for people to customise it.

Great, looking forward to it!

LukeMathWalker commented:
> > In some environments is needed (e.g. WSL2 or Docker) but I would like to avoid opening the web server to traffic coming (potentially) from the internet as a default setting.
> > Maybe as a comment?
> 
> This would be a great solution. Keep the defaults secure, but let people know that it could cause issues. That way people can be spared the debugging session

Do you want to do it updating this PR or should I create a new one?

> > From chapter03-1 it becomes configurable, so it should be easier for people to customise it.
> 
> Great, looking forward to it!

It's already in master :grin:

JesHansen commented:
> > This would be a great solution. Keep the defaults secure, but let people know that it could cause issues. That way people can be spared the debugging session
> 
> Do you want to do it updating this PR or should I create a new one?

I updated the PR, reverting the binding to localhost and added a comment explaining about the connectivity issue.

JesHansen commented:
> It's already in master 😁
Isn't it just the port number that's configurable? [Here](https://github.com/LukeMathWalker/zero-to-production/blob/24dccda00cd66110dfe83cf7539bcb94ada567f4/chapter03-1/src/main.rs#L14) it still looks like the localhost part is still hardcoded.

LukeMathWalker commented:
You are right, I'll make the host configurable as well :sweat_smile:
